concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -378,10 +379,17 @@ loadSkovPersistentData rp _treeStateDirectory pbsc = do
     GenesisBlock gd' -> return gd'
     _ -> logExceptionAndThrowTS (DatabaseInvariantViolation "Block at height 0 is not a genesis block.")
 
+  -- Unroll the treestate if the last finalized blockstate is corrupted. If the last finalized
+  -- blockstate is not corrupted, the treestate is unchanged.
+  deleteFinalizedBlocksWhile _db isBlockStateCorrupted >>= \case
+    Left e -> logExceptionAndThrowTS . DatabaseInvariantViolation $ "Database recovery failed: " <> e
+    Right bh -> logEvent GlobalState LLInfo $ "Local treestate and blockstate unrolled. Last finalized block hash is now " <> show bh
```

Comment:
This always logs a message even if nothing was removed. That is confusing, so please do not log "... unrolled ..." if nothing was removed.